tests/bdd.egg

```diff
@@ -16,81 +16,90 @@
 (rewrite (ITE n a a) a)
 
 (function and (BDD BDD) BDD)
+(rewrite (and x y) (and y x))
 (rewrite (and False n) False)
-(rewrite (and n False) False)
 (rewrite (and True x) x)
-(rewrite (and x True) x)
+
 ; We use an order where low variables are higher in tree
 ; Could go the other way.
 (rewrite (and (ITE n a1 a2) (ITE m b1 b2))
     (ITE n (and a1 (ITE m b1 b2)) (and a2 (ITE m b1 b2)))
     :when ((< n m))
 )
-(rewrite (and (ITE n a1 a2) (ITE m b1 b2))
-    (ITE m (and (ITE n a1 a2) b1) (and (ITE n a1 a2) b2))
-    :when ((> n m))
-)
 (rewrite (and (ITE n a1 a2) (ITE n b1 b2))
     (ITE n (and a1 b1) (and a2 b2))
 )
 
-(let b0 (ITE 0 True False))
-(let b1 (ITE 1 True False))
-(let b2 (ITE 2 True False))
-
-(let b123 (and b2 (and b0 b1)))
-(let b11 (and b1 b1))
-(let b12 (and b1 b2))
-(run 5)
-(query-extract b11)
-(query-extract b12)
-(query-extract b123)
-(check (= (and (ITE 1 True False) (ITE 2 True False))
-       (ITE 1 (ITE 2 True False) False))
-)
-;(check (= b123 (ITE 3 ()))
-
 (function or (BDD BDD) BDD)
+(rewrite (or x y) (or y x))
 (rewrite (or True n) True)
-(rewrite (or n True) True)
 (rewrite (or False x) x)
-(rewrite (or x False) x)
 (rewrite (or (ITE n a1 a2) (ITE m b1 b2))
```

Comment:
Note that this uses "b1" & "b2" which have been defined above in the test code. So these aren't free variables.